docs/advanced/smart_ptrs.rst

```diff
@@ -1,126 +1,113 @@
-Smart pointers
-##############
+Smart pointers & ``py::class_``
+###############################
 
-std::unique_ptr
-===============
+The binding generator for classes, ``py::class_``, can be passed a template
+type that denotes a special *holder* type that is used to manage references to
+the object. If no such holder type template argument is given, the default for
+a type ``T`` is ``std::unique_ptr<T>``.
 
-Given a class ``Example`` with Python bindings, it's possible to return
-instances wrapped in C++11 unique pointers, like so
+.. note::
 
-.. code-block:: cpp
+    A ``py::class_`` for a given C++ type ``T`` — and all its derived types —
+    can only use a single holder type.
 
-    std::unique_ptr<Example> create_example() { return std::unique_ptr<Example>(new Example()); }
 
-.. code-block:: cpp
+.. _smart_holder:
 
-    m.def("create_example", &create_example);
+``py::smart_holder``
+====================
 
-In other words, there is nothing special that needs to be done. While returning
-unique pointers in this way is allowed, it is *illegal* to use them as function
-arguments. For instance, the following function signature cannot be processed
-by pybind11.
+Starting with pybind11v3, ``py::smart_holder`` is built into pybind11. It is
+the recommended ``py::class_`` holder for most situations, but it is **not**
+the default holder, and there are no plans to make it the default holder in
+the future. This decision is based on the assumption that such a change would
```

Comment:
... this rationale is confusing. I'm reading the current wording as "smart holder is really easy to use and way better, so we decided to not make it the default".

I would split it into two paragraphs, and change the wording slightly.

> Starting with pybind11v3, ``py::smart_holder`` is built into pybind11. It is the recommended ``py::class_`` holder for most situations. For backwards compatibility, it is **not** the default holder, and there are no plans to make it the default holder in the future.
>
> It is extremely easy to use the safer and more versatile ``py::smart_holder``. ...

Reply:
Done in commit 449ccebc8a64894c45588660e32b5e11ba2885e3.